server/game/stateWatchers/CardsDiscardedThisPhaseWatcher.ts

```diff
@@ -0,0 +1,50 @@
+import { InPlayCard } from '../core/card/baseClasses/InPlayCard';
+import Player from '../core/Player';
+import { Card } from '../core/card/Card';
+import type { Location } from '../core/Constants';
+import { StateWatcher } from '../core/stateWatcher/StateWatcher';
+import { StateWatcherName } from '../core/Constants';
+import { StateWatcherRegistrar } from '../core/stateWatcher/StateWatcherRegistrar';
+
+export interface CardDiscardedEntry {
+    card: Card;
+    controlledBy: Player;
+    discardedFromLocation: Location;
+}
+
+export type ICardsDiscardedThisPhase = CardDiscardedEntry[];
```

Comment:
Let's move this file and the related changes out of this PR since it's for the Kylo PR and therefore isn't being tested. If we check it in then people will assume it's tested and working.

Reply:
Yes sir